### PR TITLE
Add azure postgres flexible module

### PR DESCRIPTION
**Pull request by mattJsonar**

### CODE REVIEW

File: examples/onboard-azure-postgresql-flexible/README.md [hunk 1]
```diff
@@ -0,0 +1,19 @@
...
+
+This example creates 'azurerm' and 'dsfhub' resources. More information regarding authentication to each can be found in the relevant provider documentation:
+- [azurerm](https://registry.terraform.io/providers/hashicorp/azurerm/latest/docs)
+- [dsfhub](https://registry.terraform.io/providers/imperva/dsfhub/latest/docs)
+
+## Prerequisites
+This module expects an Event Hub and a Storage Account Container to have been created in advance, in addition to a corresponding existing AZURE EVENTHUB asset in DSF. Both of these prerequisites and all related resources are handled in the ``onboard-azure-eventhub`` module. It also expects a method to connect to the PostgreSQL instance to create the ``pgaudit`` extension.
+
+### Azure Event Hub Namespace and Event Hub
+Cosmos DB audit logs are sent to an Azure Event Hub and are retrieved by DSF. The Event Hubs are created within an Event Hub Namespace, which can contain one or more Event Hubs. Audit logs of multiple Cosmos DB instances can be sent to a single Event Hub.
```
Comment (PetalJsonar) [inline examples/onboard-azure-postgresql-flexible/README.md:12]: Some references to Cosmos here

File: modules/dsfhub-azure-postgresql-flexible/variables.tf [hunk 2]
```diff
@@ -0,0 +1,96 @@
...
+  type        = string
+  default     = "default"
+}
+
+variable "server_host_name" {
+  description = "Hostname of the PostgreSQL instance."
+  type        = string
+}
+
+variable "server_ip" {
```
Comment (PetalJsonar) [inline modules/dsfhub-azure-postgresql-flexible/variables.tf:81]: this variable doesn't exist in main (on that note, do we want to treat `server_host_name` the same way?)

Comment (mattJsonar) [reply modules/dsfhub-azure-postgresql-flexible/variables.tf:81]: Ahh woops, think that's just my mistake. Removed it from the base module as USC doesn't expect it in the payload but forgot to propagate the change up. 

What do you mean by treating `server_host_name` the same way though?

File: modules/onboard-azure-postgresql-flexible/variables.tf [hunk 3]
```diff
@@ -0,0 +1,189 @@
...
+  default     = true
+}
+
+variable "server_resource_group_name" {
+  description = "The name of the Resource Group where the PostgreSQL Flexible Server should exist. Changing this forces a new PostgreSQL Flexible Server to be created."
+  type        = string
+}
+
+variable "server_sku_name" {
+  description = "The SKU Name for the PostgreSQL Flexible Server. The name of the SKU, follows the tier + name pattern (e.g. B_Standard_B1ms, GP_Standard_D2s_v3, MO_Standard_E4s_v3)."
```
Comment (PetalJsonar) [inline modules/onboard-azure-postgresql-flexible/variables.tf:125]: do we want to mention the default value? here and in the module itself

File: modules/dsfhub-azure-postgresql-flexible/main.tf [hunk 4]
```diff
@@ -0,0 +1,36 @@
...
+  asset_id                  = var.asset_id
+  audit_pull_enabled        = var.audit_pull_enabled
+  database_name             = var.database_name
+  gateway_id                = var.gateway_id
+  location                  = var.location
+  logs_destination_asset_id = var.logs_destination_asset_id
+  parent_asset_id           = var.parent_asset_id
+  server_host_name          = var.server_host_name
+  server_ip                 = null
+  server_port               = var.server_port
```
Comment (PetalJsonar) [inline modules/dsfhub-azure-postgresql-flexible/main.tf:23]: Is it possible to connect on a different port or can we just stick 5432 in here?

Comment (mattJsonar) [reply modules/dsfhub-azure-postgresql-flexible/main.tf:23]: No, doesn't appear to let you choose. I'll remove it.

**Review by PetalJsonar — APPROVED** [review]

**Pull request merged**